infra/aws/terraform/prow-build-cluster/resources/karpenter/nodepool-default.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: default
+spec:
+  template:
+    spec:
+      nodeClassRef:
+        apiVersion: karpenter.k8s.aws/v1beta1
+        kind: EC2NodeClass
+        name: default
+
+      requirements: # r5ad.4xlarge amd64 non-spot
+        - key: topology.kubernetes.io/zone
+          operator: In
+          values: ["us-east-2a", "us-east-2b", "us-east-2c"]
+        - key: kubernetes.io/arch
+          operator: In
+          values: ["amd64"]
```

Comment:
do we need to be as restrictive on some of these? if folks want to run arm, why not let them?

likewise below for the `instance-*`, why not let Karpenter decide the best fit?

Reply:
We could have a dedicated nodepool with arm64 instances ?

Reply:
At the moment yes, we don't want to change the behavior now, this needs a bit more discussion with SIG Testing.

Reply:
I guess what I am saying is why block or restrict things that require users to come ask us to unblock. If a group wanted to test arm64 instances, it's seems like we should just let them do that. And on the instance types/sizes - Karpenter works better (delivers on its benefits) when it's not restricted. The workloads define their requirements and Karpenter supplies compute to satisfy those requirements

Reply:
I'm +1 with this, but the plan is that we try to replicate the cluster-autoscaler behavior in the very beginning, and then we start to add additional functionalities after the initial deployment.